test/types/tuple/heterogeneous/loops/forLoopHetTuple.chpl

```diff
@@ -0,0 +1,18 @@
+record R {
+  var x = 42;
+
+  proc init=(other: R) {
+    writeln("In copy init");
+    this.x = other.x + 1;
+  }
+}
+
+proc =(ref lhs: R, rhs: R) {
+  writeln("In assignment");
+  lhs.x = rhs.x + 5;
+}
+
+var tup = (1, 2.3, "hi", new R());
+
+for t in tup do
+  writeln(t);
```

Comment:
Seems like you need some for loops with `continue` and `break` as well

Reply:
Good point.  I've added tests in https://github.com/chapel-lang/chapel/pull/15034/commits/fa5efb593f7e6a5c652ddfbcfae65409f213da63 and 00aa91d630848e91b1faac1b30bdbe55b0c3f73f for these cases.